compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2775,7 +2775,11 @@ extends SyntaxMsg(TargetNameOnTopLevelClassID):
 class NotClassType(tp: Type)(using Context)
 extends TypeMsg(NotClassTypeID), ShowMatchTrace(tp):
   def msg(using Context) = i"$tp is not a class type"
-  def explain(using Context) = ""
+  def explain(using Context) =
+    i"""A class type includes classes and traits in a specific order. Defining a class, even an anonymous class,
+        |requires specifying an order for the traits it extends. For example, `A & B` is not a class type because it
+        |doesn't specify which trait takes precedence, A or B. Both `A with B` and `B with A` are class types.
```

Comment:
In Scala 3 you can also write ``class Foo extends A, B` instead of `extends A with B`, it might be simpler to speak about a list of class types instead.